src/sortedDict.jl

```diff
@@ -79,336 +58,50 @@ end
-keys(m::SortedDict) = SDKeyIteration(m)
-values(m::SortedDict) = SDValIteration(m)
-keys{T <: SDIterableTypesBase}(ba::T) = SDKeyIteration(ba)
-values{T <: SDIterableTypesBase}(ba::T) = SDValIteration(ba)
-tokens{T <: SDIterableTypesBase}(ba::T) = SDTokenIteration(ba)
-tokens{T <: SDIterableTypesBase}(ki::SDKeyIteration{T}) = SDTokenKeyIteration(ki.base)
-tokens{T <: SDIterableTypesBase}(vi::SDValIteration{T}) = SDTokenValIteration(vi.base)
-
-start(m::SortedDict) = SDIterationState(m, nextloc0(m.bt,1), 2)
-
-function start(e::ExcludeLast) 
-    (!(e.first in e.m.bt.useddatacells) || e.first == 1 ||
-        !(e.pastlast in e.m.bt.useddatacells)) &&
-        throw(BoundsError())
-    if compareInd(e.m.bt, e.first, e.pastlast) < 0
-        return SDIterationState(e.m, e.first, e.pastlast) 
-    else
-        return SDIterationState(e.m, 2, 2)
-    end
-end
-
-function start(e::IncludeLast) 
-    (!(e.first in e.m.bt.useddatacells) || e.first == 1 ||
-        !(e.last in e.m.bt.useddatacells) || e.last == 2) && 
-        throw(BoundsError())
-    if compareInd(e.m.bt, e.first, e.last) <= 0
-        return SDIterationState(e.m, e.first, nextloc0(e.m.bt, e.last)) 
-    else
-        return SDIterationState(e.m, 2, 2)
-    end
-end
-
-start(e::SDCompoundIterable) = start(e.base)
-
-function isless{K, D, Ord <: Ordering}(s::SDToken{K,D,Ord}, t::SDToken{K,D,Ord})
-    !(s.container === t.container) &&  
-        throw(ArgumentError("SDToken isless requires tokens for the same container"))
-    return compareInd(s.container.bt, 
-                      s.semitoken.address, 
-                      t.semitoken.address) < 0
-end
-
-
-function isequal{K, D, Ord <: Ordering}(s::SDToken{K,D,Ord}, t::SDToken{K,D,Ord})
-    !(s.container === t.container) && 
-        throw(ArgumentError("SDToken isequal requires tokens for the same container"))
-    return s.semitoken.address == t.semitoken.address
-end
-
-
-function in{K,D,Ord <: Ordering}(pr::(@compat Tuple{Any,Any}), m::SortedDict{K,D,Ord})
+@inline function in{K,D,Ord <: Ordering}(pr::(@compat Tuple{Any,Any}), m::SortedDict{K,D,Ord})
     i, exactfound = findkey(m.bt,convert(K,pr[1]))
     return exactfound && isequal(m.bt.data[i].d,convert(D,pr[2]))
 end
 
-eltype{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}) = (K,D)
+@inline eltype{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}) = @compat Tuple{K,D}
+@inline orderobject(m::SortedDict) = m.bt.ord
+@inline keytype{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}) = K
+@inline datatype{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}) = D
 
-orderobject(m::SortedDict) = m.bt.ord
 
-function haskey{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}, k_)
-    i, exactfound = findkey(m.bt,convert(K,k_))
+@inline function haskey(m::SortedDict, k_)
+    i, exactfound = findkey(m.bt,convert(keytype(m),k_))
     exactfound
 end
 
-function get{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}, k_, default_)
+@inline function get{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}, k_, default_)
```

Comment:
I implicitly pointed it out above, but since you don't use `Ord` in the function, you can leave it off (which reduces noise).